### ISSUE TITLE
should be clearer about where the definition of the HTTP Protocol Binding is

### ISSUE THREAD

**Issue by dbaron** [w3c/wot-thing-description#791]
There are various parts of the thing description specification that reference the HTTP Protocol Binding.  Most prominently, there's a section [8.3.1 HTTP Protocol Binding](https://w3c.github.io/wot-thing-description/#http-binding-assertions).  This talks a bit about the HTTP Protocol Binding, but there isn't clearly either (a) a definition of the HTTP Protocol Binding or (b) a link to such a definition.  It seems like there should be one or the other.

In other words, it seems like either the section should clearly say that it's defining the HTTP Protocol Binding and what the HTTP Protocol Binding is, or it should link to something else that does.  (I don't *think* the current section has enough to define it -- but maybe I'm missing something in that I don't know exactly what part of the system constitutes the HTTP Protocol Binding.)

(I got here from w3ctag/design-reviews#357.)

**Comment by takuki** [comment]
The WoT Thing Description TF discussed this issue during 7/12 telecon.
Minutes is available [here](https://www.w3.org/2019/07/12-wot-td-minutes.html#item012).

Excerpt from minutes:
> TD Issue 791
> Taki: (It was pointed out that) section 8.3.1 protocol binding is not defined. What is a protocol binding?
> Sebastian:  Also (it) could be more well defined in the binding template document.
> Kaz: "Protocol Binding" itself (is) defined in the architecture document.
> Kaz: We could explain HTTP as it is the most popular protocol.
> Sebastian: We can also point out that we are using existing communication metadata vocabularies.
> Kaz: We can describe protocol binding as similar to a language binding mechanism for i18n.

**Comment by sebastiankb** [comment]
The Section 8.3 Protocol Bindings provides an introduction what is the basic intention of the protocol binding and that can applied to different protocols like HTTP, CoAP and MQTT.
The Section 8.3.1 explains how the protocol binding is applied to http which is the default protocol assumption in the TD. We provided a PR #828 that made some changes in wording to make it more clear that is no different definition 'HTTP Protocol Binding' it is more the basic definition 'Protocol Binding' that is applied to http.

**Comment by benfrancis** [comment]
I'm still not sure this is entirely clear.

By my understanding there seem be several parts to what the WoT specifications describe as a "protocol binding":
1. A vocabularly for describing an existing underlying protocol e.g. https://www.w3.org/TR/HTTP-in-RDF10/
2. "Further details" for how to define protocol bindings in the Protocol Binding Template specification https://www.w3.org/TR/wot-binding-templates/
3. "Hypermedia controls" (forms) inside each instance of a Thing Description https://w3c.github.io/wot-thing-description/#form which declaratively define methods, headers and payloads for each "interaction affordance" (though it has been acknowledged are not currently expressive enough to describe any existing web API/protocol)
4. "Defaults" in the Thing Description specification which define a set of default protocol methods to associate with (a subset of) "interaction affordances" if none are provided in a Thing Description instance via forms https://w3c.github.io/wot-thing-description/#http-binding-assertions

And in future:
5. A proposed additional "interoperability profiles" mechanism which can also define protocol bindings e.g. https://w3c.github.io/wot-profile/#http-protocol-binding

I suggest that this could all be drastically simplified to:
1. A single specification for each protocol binding (or perhaps more accurately "sub-protocol") which describes how to communicate with a web thing over a given protocol (e.g. https://iot.mozilla.org/wot/#web-thing-rest-api, https://iot.mozilla.org/wot/#web-thing-websocket-api or https://w3c.github.io/wot-profile/#http-protocol-binding)
2. A mechanism in the Thing Description specification to reference the sub-protocol used by a given web thing

With this approach parts 1-4 above would be unnecessary and part 5 could instead form the basis of the mechanism for referring to an external protocol binding/sub-protocol specification.

I acknowledge that these simplifications are unlikely in this round of the specification publication and that there are still many members who like the declarative "hypermedia controls" approach. Therefore in the meantime is it possible that the specification could better explain the relationship between parts 1-4 above?

**Comment by mmccool** [comment]
Ben,
Thanks for your input.  I agree the definition of “Protocol Binding” needs to be clearer and more compact.   However, we need to make a CR transition request today if we are to meet the schedule for PR transition in our current charter.  Do you consider this a blocking issue at this point?   Based on my reading of your comments, I’m going to assume it’s not, and will proceed with the transition request, and mark this issue as “by PR” but please let us know otherwise ASAP.

We can still make informative changes prior to PR to better explain 1-4 as you suggest, then address this more completely with normative revisions in our update planned for the next charter period.   So the “by PR” part would address the informative revisions, then we probably should make a second issue for the second, “by next TD revision”, part.

**Comment by egekorkan** [comment]
For a comment regarding how it can be simplified later on:
* Protocol Binding is information contained in a Form (Thing or Interaction level) that describes how the `op` keyword of this form can be implemented/used with a specific (application layer) protocol.

So a long explanation would be: 
* For an interaction, everything but the contents of forms array is protocol independent. With the content of `forms`, one can describe how this interaction can used with different protocols. The consumer of the TD, should look for the `op` keyword in the `forms` of this interaction that it is interested in based on the application logic and should construct the protocol request based on the information contained in the form with the desired `op` keyword. This way, the Consumer binds the application logic to a concrete protocol, thus the term **Protocol Binding**.

In the binding document, there is also the adaptation of payload for different `contentType`s but this does not seem to be the issue here.
